Meshtastic/Helpers/BLEManager.swift

```diff
@@ -834,45 +834,92 @@ class BLEManager: NSObject, CBPeripheralDelegate, MqttClientProxyManagerDelegate
-							connectedHop.longitudeI = mostRecent.longitudeI
-							traceRoute?.hasPositions = true
+						let traceRouteHop = TraceRouteHopEntity(context: context)
+						traceRouteHop.time = Date()
+						if routingMessage.snrTowards.count >= index + 1 {
+							traceRouteHop.snr = Float(routingMessage.snrTowards[index]) / 4
+						} else {
+							// If no snr in route, set unknown
+							traceRouteHop.snr = -32
+						}
+						if let hn = hopNode, hn.hasPositions {
+							if let mostRecent = hn.positions?.lastObject as? PositionEntity, mostRecent.time! >= Calendar.current.date(byAdding: .hour, value: -24, to: Date())! {
+								traceRouteHop.altitude = mostRecent.altitude
+								traceRouteHop.latitudeI = mostRecent.latitudeI
+								traceRouteHop.longitudeI = mostRecent.longitudeI
+								traceRoute?.hasPositions = true
+							}
 						}
-						var routeString = "\(connectedNode.user?.longName ?? "???") --> "
-						hopNodes.append(connectedHop)
-						traceRoute?.hopsTowards = Int32(routingMessage.route.count)
-						for (index, node) in routingMessage.route.enumerated() {
+						traceRouteHop.num = hopNode?.num ?? 0
+						if hopNode != nil {
+							if decodedInfo.packet.rxTime > 0 {
+								hopNode?.lastHeard = Date(timeIntervalSince1970: TimeInterval(Int64(decodedInfo.packet.rxTime)))
+							}
+						}
+						hopNodes.append(traceRouteHop)
+
+						let hopName = hopNode?.user?.longName ?? (node == 4294967295 ? "Repeater" : String(hopNode?.num.toHex() ?? "unknown".localized))
+						let mqttLabel = hopNode?.viaMqtt ?? false ? "MQTT " : ""
+						let snrLabel = (traceRouteHop.snr != -32) ? String(traceRouteHop.snr) : "unknown ".localized
+						routeString += "\(hopName) \(mqttLabel)(\(snrLabel)dB) --> "
+					}
+					let destinationHop = TraceRouteHopEntity(context: context)
+					destinationHop.name = traceRoute?.node?.user?.longName ?? "unknown".localized
+					destinationHop.time = Date()
+					// If nil, set to unknown, INT8_MIN (-128) then divide by 4
+					destinationHop.snr = Float(routingMessage.snrTowards.last ?? -128) / 4
+					destinationHop.num = traceRoute?.node?.num ?? 0
+					if let mostRecent = traceRoute?.node?.positions?.lastObject as? PositionEntity, mostRecent.time! >= Calendar.current.date(byAdding: .hour, value: -24, to: Date())! {
+						destinationHop.altitude = mostRecent.altitude
+						destinationHop.latitudeI = mostRecent.latitudeI
+						destinationHop.longitudeI = mostRecent.longitudeI
+						traceRoute?.hasPositions = true
+					}
+					hopNodes.append(destinationHop)
+					/// Add the destination node to the end of the route towards string and the beginning of the route back string
+					routeString += "\(traceRoute?.node?.user?.longName ?? "unknown".localized) \((traceRoute?.node?.num ?? 0).toHex()) (\(destinationHop.snr != -32 ? String(destinationHop.snr) : "unknown ".localized)dB)"
+					traceRoute?.routeText = routeString
+
+					traceRoute?.hopsBack = Int32(routingMessage.routeBack.count)
```

Comment:
@bjpetit I believe it now shows "Direct Back" when the `if` statement below is false (`hopStart` is not set), which is a bit misleading.

Reply:
@GUVWAF I've posted #1046 to try and address this. Thanks for the note.